sbt-ash/src/main/scala/org/virtuslab/ash/AkkaSerializationHelperPlugin.scala

```diff
@@ -0,0 +1,65 @@
+package org.virtuslab.ash
+
+import sbt.Def
+import sbt.Keys._
+import sbt._
+
+object AkkaSerializationHelperPlugin extends AutoPlugin {
+
+  object autoImport extends AkkaSerializationHelperKeys {
+    val baseAshSettings: Seq[Def.Setting[_]] = AkkaSerializationHelperPlugin.baseAshSettings
+  }
+  import autoImport.{baseAshSettings => _, _}
+
+  override def projectSettings: Seq[Def.Setting[_]] = additionalSettings ++ ashSettings
+
+  override def globalSettings: Seq[Def.Setting[_]] = Nil
+
+  lazy val additionalSettings: Seq[Def.Setting[_]] = Seq(
+    libraryDependencies ++= Seq(
+        compilerPlugin(ashDumpEventSchemaCompilerPlugin.value),
+        compilerPlugin(ashRegistrationCheckerCompilerPlugin.value),
+        compilerPlugin(ashSerializabilityCheckerCompilerPlugin.value)),
+    scalacOptions ++= Seq(
+        s"-P:dump-event-schema-plugin:${target.value}",
+        s"-P:registration-checker-plugin:${target.value}"),
+    cleanFiles ++= Seq(
+        target.value / "dump-event-schema-cache",
+        target.value / "registration_checker_plugin_cache.csv"),
+    scalacOptions ++= {
+      val de = (ashDumpEventSchemaCompilerPlugin / ashCompilerPluginEnable).value
+      val dv = (ashDumpEventSchemaCompilerPlugin / ashCompilerPluginVerbose).value
+      val re = (ashRegistrationCheckerCompilerPlugin / ashCompilerPluginEnable).value
+      val rv = (ashRegistrationCheckerCompilerPlugin / ashCompilerPluginVerbose).value
+      val se = (ashSerializabilityCheckerCompilerPlugin / ashCompilerPluginEnable).value
+      val sv = (ashSerializabilityCheckerCompilerPlugin / ashCompilerPluginVerbose).value
+      val l1 = List(de, re, se).map { x =>
+        if (x) "" else "--disable"
+      }
+      val l2 = List(dv, rv, sv).map { x =>
+        if (x) "--verbose" else ""
+      }
+      val plugins = List("dump-event-schema-plugin", "registration-checker-plugin", "serializability-checker-plugin")
+
+      l1.zip(l2).zip(plugins).flatMap { x =>
+        val ((a, b), plugin) = x
+        Seq(a, b).filter(_.nonEmpty).map(x => s"-P:$plugin:$x")
+      }
+    })
+
+  lazy val ashSettings: Seq[Def.Setting[_]] = baseAshSettings
+
+  lazy val baseAshSettings: Seq[Def.Setting[_]] = Seq(
+    ashDumpEventSchemaCompilerPlugin := "org.virtuslab" %% "dump-event-schema-compiler-plugin" % (ashDumpEventSchemaCompilerPlugin / ashCompilerPluginVersion).value,
+    ashRegistrationCheckerCompilerPlugin := "org.virtuslab" %% "registration-checker-compiler-plugin" % (ashRegistrationCheckerCompilerPlugin / ashCompilerPluginVersion).value,
+    ashSerializabilityCheckerCompilerPlugin := "org.virtuslab" %% "serializability-checker-compiler-plugin" % (ashSerializabilityCheckerCompilerPlugin / ashCompilerPluginVersion).value,
```

Comment:
AFAICS (testing on Hydra branch `chore/ash-single-sbt-plugin`) I still need to add the following:

```scala
    "org.virtuslab"                 %% "circe-akka-serializer"             % Versions.akkaSerializationHelper,
    "org.virtuslab"                 %% "serializability-checker-library"   % Versions.akkaSerializationHelper
```
explicitly.

Following approach taken by [Play sbt plugin (link)](https://github.com/playframework/playframework/blob/master/dev-mode/sbt-plugin/src/main/scala/play/sbt/PlayImport.scala), I'd make `circe-akka-serializer` available under an alias identifier... like `circeAkkaSerializer`.

Not sure what to do with `serializability-checker-library` (aka `ash-annotation`)... maybe just add automatically to the classpath of every module where `registration-checker-compiler-plugin` or `serializability-checker-compiler-plugin` is enabled (i.e. is NOT `--disable`d)?

Reply:
> Not sure what to do with serializability-checker-library (aka ash-annotation)... maybe just add automatically to the classpath of every module 

Better not. This would be magic in the background that the end user would have to understand.

Reply:
>AFAICS (testing on Hydra branch chore/ash-single-sbt-plugin) I still need to add the following:

I'm not sure the Play approach would fix this thing considering that dependencies are declared in `project/Dependencies`. This would require adding this plugin in `project/project/plugin.sbt`.

Reply:
Hmmm but doesn't project/Dependencies.scala have access to the very same stuff as build.sbt?

Reply:
As in, project/build.sbt or project/project/Dependencies.scala are on a deeper level, but project/Dependencies.scala should see stuff provided by sbt plugins defined in project/plugins.sbt IIUC

Reply:
Ok, implemented at Hydra

Reply:
:benissimo: